driver/normalizer/normalizer.go

```diff
@@ -772,3 +770,180 @@ func (op dropNils) Check(st *State, n nodes.Node) (bool, error) {
 func (op dropNils) Construct(st *State, n nodes.Node) (nodes.Node, error) {
 	return op.op.Construct(st, n)
 }
+
+var (
+	typeGroup     = uast.TypeOf(uast.Group{})
+	typeFuncGroup = uast.TypeOf(uast.FunctionGroup{})
+)
+
+// triviaField specified a field with an array to put trivias into.
+var triviaField = map[string]string{
+	"Block":           "Statements",
+	"CompilationUnit": "Members",
+}
+
+// opMoveTrivias cuts trivia nodes from LeadingTrivia/TrailingTrivia fields
+// and wraps the node into uast:Group that contains those trivias.
+type opMoveTrivias struct {
+	sub Op
+}
+
+func (op opMoveTrivias) Kinds() nodes.Kind {
+	return nodes.KindObject
+}
+
+func (op opMoveTrivias) Check(st *State, n nodes.Node) (bool, error) {
+	obj, ok := n.(nodes.Object)
+	if !ok {
+		return false, nil
+	}
+	cloned := false
+	leading, ok1 := obj["LeadingTrivia"].(nodes.Array)
+	trailing, ok2 := obj["TrailingTrivia"].(nodes.Array)
+	if ok1 || ok2 {
+		// we saved the leading and trailing trivias, now wipe them from the node
+		obj = obj.CloneObject()
+		cloned = true
+		delete(obj, "LeadingTrivia")
+		delete(obj, "TrailingTrivia")
+	}
+
+	// move comments out of keywords and tokens into actual AST nodes
+	// TODO(dennwc): handle more cases:
+	//				 - Modifiers?
+	for key, sub := range obj {
+		if key != "ReturnType" &&
+			!strings.HasSuffix(key, "Token") &&
+			!strings.HasSuffix(key, "Keyword") {
+			continue
+		}
+		if uast.TypeOf(sub) != typeGroup {
+			continue
+		}
+		group := sub.(nodes.Object)
+		arr, ok := group["Nodes"].(nodes.Array)
+		if !ok {
+			continue
+		}
+		// we already joined trivias and the node into a single array,
+		// so we will have to find its index now
+		ind := -1
+		for i, sub := range arr {
+			if !strings.HasSuffix(uast.TypeOf(sub), "Trivia") {
+				ind = i
+				break
+			}
+		}
+		var node nodes.Node
+		if ind < 0 {
+			// cannot determine an index - pretend everything is a leading trivia
+			// TODO(dennwc): find another way if it happens in practice
```

Comment:
"another"? Is there a particular case where it happens now?

Reply:
I don't know cases when it might happen right now. Clarified the comment to mention it.